pkg/controller/node/nodecontroller_test.go

```diff
@@ -2060,6 +2062,186 @@ func TestSwapUnreachableNotReadyTaints(t *testing.T) {
 	}
 }
 
+func TestTaintsNodeByCondition(t *testing.T) {
+	fakeNow := metav1.Date(2017, 1, 1, 12, 0, 0, 0, time.UTC)
+	evictionTimeout := 10 * time.Minute
+
+	fakeNodeHandler := &testutil.FakeNodeHandler{
```

Comment:
In this test, it only care about the update of one node status; it's not necessary to build `fakeNodeHandler` with different existing nodes :).